kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -598,8 +610,11 @@ private void convertRule(Rule rule, boolean heatCoolEq, Sort topCellSort, Map<St
                     rule = rule.withAtt(rule.att().add(Attribute.SIMPLIFICATION_KEY));
                 }
             }
+            sb.append("  axiom{R");
+            StringBuilder oldSB = sb;
```

Comment:
This is very weird and not nice. What are you trying to accomplish here?

Reply:
I agree this is very weird and not nice, but it's because one might need to add extra sort variables which we don't know of until we go through the terms.

I guess we could be traversing the terms here and collecting the sort variables; that would be less weird, although not very nice as well.

Reply:
I would rather annotate the rule with an attribute containing the sort variables it needs when those variables are created.